src/components/TextEmoji.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import _ from 'underscore';
+import Str from 'expensify-common/lib/str';
+import Text from './Text';
+import * as EmojiUtils from '../libs/EmojiUtils';
+import * as StyleUtils from '../styles/StyleUtils';
+import stylePropTypes from '../styles/stylePropTypes';
+
+const propTypes = {
+    /** The message text to render */
+    children: PropTypes.string.isRequired,
+
+    /** The message text additional style */
+    style: stylePropTypes,
+};
+
+const defaultProps = {
+    style: [],
+};
+
+const TextEmoji = (props) => {
+    if (!EmojiUtils.containsEmoji(props.children)) {
+        return (
+            <Text>
+                {props.children}
+            </Text>
+        );
+    }
+
+    const propsStyle = StyleUtils.parseStyleAsArray(props.style);
+
+    return _.map([...Str.htmlDecode(props.children)], ((char, index) => (
```

Comment:
So this component applies style to only the emoji characters in a string. So given a string like this:

```
smile U+1F600!
```

We would end up with a component hierarchy like this (ignoring keys):

```jsx
<>
    <Text>s</Text>
    <Text>m</Text>
    <Text>i</Text>
    <Text>l</Text>
    <Text>e</Text>
    <Text>{' '}</Text>
    <View>
        <Text style={propsStyle}>
            U+1F600
        </Text>
    </View>
    <Text>!</Text>
</>
```

But I think we could be smarter and land ourselves with a leaner component hierarchy like this:

```jsx
<>
    <Text>{'smile '}</Text>
    <View>
        <Text style={propsStyle}>
            U+1F600
        </Text>
    </View>
    <Text>!</Text>
</>
```

To do that, I think we could split the string using the emoji regex, rather than just mapping over all characters. The only thing is that we need to wrap the emoji regex in a capture group.

So given this input string:

```
smile U+1F600!
```

You'd want to create an `EmojiUtils` function called `splitStringByEmojis` that produces this as output:

```js
[
    {text: 'smile ', containsEmoji: false},
    {text: 'U+1F600', containsEmoji: true},
    {text: '!', containsEmoji: false},
]
```

Then in here you could loop over that array rather than the string.

Reply:
Also, I think it would be a good idea to write some unit tests to cover that function